src/runInContext.js

```diff
@@ -0,0 +1,235 @@
+var attempt = require("./attempt"),
+    makeUTCDate = require("./date"),
+    makeForOwn = require("./forOwn"),
+    makeParse = require("./parse"),
+    makeSerializeDate = require("./serializeDate"),
+    makeStringify = require("./stringify");
+
+module.exports = makeRunInContext;
+function makeRunInContext(root) {
+  // Public: Initializes JSON 3 using the given `context` object, attaching the
+  // `stringify` and `parse` functions to the specified `exports` object.
+  function runInContext(context, exports) {
+    context || (context = root.Object());
+    exports || (exports = root.Object());
+
+    // Native constructor aliases.
+    var String = context.String || root.String,
+        Object = context.Object || root.Object,
+        Date = context.Date || root.Date,
+        SyntaxError = context.SyntaxError || root.SyntaxError,
+        TypeError = context.TypeError || root.TypeError,
+        Math = context.Math || root.Math,
+        nativeJSON = context.JSON || root.JSON;
+
+    // Delegate to the native `stringify` and `parse` implementations.
+    if (typeof nativeJSON == "object" && nativeJSON) {
+      exports.stringify = nativeJSON.stringify;
+      exports.parse = nativeJSON.parse;
+    }
+
+    // Convenience aliases.
+    var getClass = Object.prototype.toString;
+
+    // Internal: Determines whether the native `JSON.stringify` and `parse`
+    // implementations are spec-compliant. Based on work by Ken Snyder.
+    function has(name) {
```

Comment:
Need to check if [`r.js`](http://requirejs.org/docs/optimization.html#hasjs) can still work with this...